File: leetcode/questions/2554/python/2554.py
from typing import List

class Solution:
    def maxCount(self, banned: List[int], n: int, maxSum: int) -> int:
        banned = {num for num in banned if num <= n}
        cur_sum = 0
        cur_cnt = 0

        for num in range(1, n + 1):
            if num + cur_sum > maxSum:
                break
            if num not in banned:
                cur_sum += num
                cur_cnt += 1
        
        return cur_cnt

File: leetcode/questions/2554/cpp/2554.cpp
#include <vector>
#include <set>

class Solution {
public:
    int maxCount(std::vector<int>& banned, int n, int maxSum) {
        std::set<int> setBanned;
        int curSum = 0;
        int cnt = 0;

        for (int num: banned) {
            if (num <= n) {
                setBanned.insert(num);
            }
        }

        for (int num = 1; num <= n && num + curSum <= maxSum; ++num) {
            if (setBanned.find(num) == setBanned.end()) {
                curSum += num;
                cnt += 1;
            }
        }

        return cnt;
    }
};